dvc/fs/fsspec_wrapper.py

```diff
@@ -215,3 +216,44 @@ def find(self, path_info, detail=False, prefix=None):
             return None
 
         yield from self._strip_buckets(files, detail=detail)
+
+
+_LOCAL_FS = LocalFileSystem()
+
+
+class CallbackMixin:
```

Comment:
I think it's a good opportunity to consider getting rid of progress bars from the filesystem and lifting these up. Related: #6262

Reply:
Interesting, though I don't see anything related to 'where to lift' these progress. Mind explaining it a bit further?

Reply:
Looking at it from the CLI, I expect these *updates* to come from each component that I am interacting with, rather than filesystem creating progress bars itself. So, I think the responsibility falls on individual components like `odb`/`checkout`/`repro`/`experiments` to provide these. These days I am leaning more and more towards the event design patterns (similar to [`pyee`](https://pyee.readthedocs.io/en/latest/)) with observable/pub-sub.
The decoupling that we will is good, but the debugging gets harder, it's a tradeoff.

I do think that the solutions for all of these components might differ based on their nature, so I would love to hear out proposals for #6262 for `odb` and friends. Not trying to block your PR, it looks good.